build.gradle

```diff
@@ -148,6 +148,7 @@ dependencies {
         exclude module: "json"
     }
     compile 'com.fasterxml.jackson.datatype:jackson-datatype-joda:2.4.0'
+    compile 'com.grack:nanojson:1.1'
```

Comment:
Please don't take this comment about dependencies to serious, my point was that this validation is completely possible with the existing dependencies and without parsing the string multiple times.

Both jackson and org.json support reading multiple objects from the same stream, so don't consider trailing characters after the first object as invalid. As a user of those libraries it is easy to validate that the whole input has been consumed parsing from a `JSONParser` or `JSONTokener`. The following method passes nearly all unit tests you have below, the only exception being that org.json allows trailing commas after a key-value pair (noted as n_object_trailing_comma.json in http://seriot.ch/json/parsing.html#24).

    private void validate(String input) {
        final JSONTokener tokener = new JSONTokener(input);
        try {
            final JSONObject json = new JSONObject(tokener);
        } catch (JSONException e) {
            throw new InvalidEventTypeException("Invalid json", e);
        }
        if (tokener.nextClean() != 0) {
            throw new InvalidEventTypeException("Invalid content after json");
        }
    }

This simple validation could be inlined and you could create the schema from the `JSONObject` without having to parse the input again.